hubblestack/hec/opt.py

```diff
@@ -0,0 +1,94 @@
+
+# NOTE: this module receives __salt__, __grains__, etc from daemon.py during refresh_grains
+
+# there's a lot of support below for things like this:
+#   get_splunk_options('hubblestack:returner:splunk', 'hubblestack:nebula:returner:splunk')
+#
+# which looks in 'hubblestack:nebula:returner:splunk' iff nothing is found in 'hubblestack:returner:splunk'
+#
+# and support for things like this:
+#   get_splunk_options('hubblestack:returner:splunk', some_default='something')
+#
+# which adds some_default to the default values
+#
+# but if nothing else is specified,
+#   list_of_opts = get_splunk_options()
+#
+# we just look in [config.get]('hubblestack:returner:splunk')
+
+import copy
+
+class Required(object):
+    pass
+REQUIRED = Required()
+del Required
+
+MODALITIES = ('config.get',) # used to house grains.get before config.get
+
+def _get_splunk_options(space, modality, **kw):
+    ret = list()
+
+    base_opts = {
+        'token': REQUIRED,
+        'indexer': REQUIRED,
+        'index': REQUIRED,
+        'port': '8088',
+        'custom_fields': [],
+        'sourcetype': 'hubble_log',
```

Comment:
Unless I've missed something, this module doesn't handle sourcetype correctly. In each of the `_get_options()` that were collapsed into this one, they looked at specific config values for their sourcetype. For example, in the nova returner:

```
            processed['sourcetype'] = opt.get('sourcetype_nova', 'hubble_audit')
```

I don't think this will be handled correctly in the new code.

Reply:
Turns out I did notice this earlier. Might be worth going through and reviewing any "unresolved" comments of mine.

Reply:
I think I mentioned elsewhere... I did think to handle this, but forgot to actually put the code in the modules. I came up with a poorly named (I couldn't think of a good name) strategy for remaping name-prefix options to their proper name...

... I've added a commit to populate those correctly.

Reply:
Still doesn't work:

```
[DEBUG   ] Options: {"index": "default_hubble", "sourcetype": "hubble_log", "http_event_server_ssl": true, "index_extracted_fields": [], "indexer": "<redacted>", "token": "<redacted>", "proxy": {}, "timeout": 9.05, "port": "8088", "custom_fields": [], "http_event_collector_ssl_verify": true}
```

Same result for nebula except there's randomly a new field:

```
[DEBUG   ] Options: {"index": "default_hubble", "timeout": 9.05, "sourcetype": "hubble_log", "http_event_server_ssl": true, "index_extracted_fields": [], "indexer": "<redacted>", "token": "<redacted>", "add_query_to_sourcetype": true, "proxy": {}, "sourcetype_nebula": "hubble_osquery", "port": "8088", "custom_fields": [], "http_event_collector_ssl_verify": true}
```

Reply:
The new field is probably from the fact that the three modules have slightly different kwargs [in your commit](https://github.com/hubblestack/hubble/pull/277/commits/23a67e820cc5f6d8eeee0c74b7e921b71f12db26)

Reply:
well, that's what the _nick mapping is for... it remaps those slightly different names to the right name. It's totally working for me. still investigating though.

Reply:
Worked with @jettero over slack and this is now working.